File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin(), points.end(), [](const vector<int>& a, const vector<int>& b){
            return a[1] < b[1];
        });
        int x = points[0][1];
        int res = 0;
        for(int i=1;i<points.size();i++){
            if(points[i][0] > x || x > points[i][1]){
                res++;
                x = points[i][1];  
            }
        }
        return res + 1;
    }
};

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        int count = 0;
        if(points.length == 1) return 1;
        Arrays.sort(points, (a, b) -> Integer.compare(a[0], b[0]));
        int end = points[0][1];
        for(int i=1;i<points.length;i++){
            if(points[i][0] <= end){
                end = Math.min(end, points[i][1]);
            }else {
                count++;
                end = points[i][1];
            }
        }
        count++;
        return count;
    }
}